lib/compilation-queue.ts

```diff
@@ -56,15 +61,15 @@ export class CompilationQueue {
         return new CompilationQueue(ceProps('maxConcurrentCompiles', 1), ceProps('compilationEnvTimeoutMs'));
     }
 
-    enqueue(job) {
-        const enqueueAsyncId = executionAsyncId();
+    enqueue<Result>(job: Job<Result>): PromiseLike<Result> {
+        const enqueueAsyncId: number = executionAsyncId();
```

Comment:
Do we need the `: number` specification here and below?

Reply:
Is this the equivalent of "almost always auto" in C++? if calling a simple function the style should be to say "use whatever it returned" ?

Reply:
I see what you're getting at. I think in this case personally I wouldn't annotate the type because there's nothing special about it being a number here, it could be a string and work just fine (as long as `_running` is changed to a `Set<string>`). But either way is good with me, this is just a small quibble!